client/my-sites/add-ons/utils/get-feature-slug.ts

```diff
@@ -0,0 +1,27 @@
+import {
+	PRODUCT_WPCOM_UNLIMITED_THEMES,
+	PRODUCT_WPCOM_CUSTOM_DESIGN,
+	PRODUCT_NO_ADS,
+	WPCOM_FEATURES_PREMIUM_THEMES,
+	WPCOM_FEATURES_CUSTOM_DESIGN,
+	WPCOM_FEATURES_NO_ADVERTS,
+} from '@automattic/calypso-products';
+
+/**
+ * Return the feature slug relevant to a given product.
+ * (features do/need not reflect products, in principle)
+ */
+const getFeatureSlug = ( productSlug: string ) => {
+	switch ( productSlug ) {
+		case PRODUCT_WPCOM_UNLIMITED_THEMES:
+			return WPCOM_FEATURES_PREMIUM_THEMES;
+		case PRODUCT_WPCOM_CUSTOM_DESIGN:
+			return WPCOM_FEATURES_CUSTOM_DESIGN;
+		case PRODUCT_NO_ADS:
+			return WPCOM_FEATURES_NO_ADVERTS;
+		default:
+			return null;
+	}
+};
+
+export default getFeatureSlug;
```

Comment:
It's the same thing in concept. `getFeatureSlug` can be extended to return an object/list of feature slugs. I can work something else out if that method feels obtrusive or confusing in any way. The intention was to avoid overloading the card & grid components with additional props or data (which was our initial take on this), but I realise the confusion this mapping creates when contained in a separate function. It's still a mapping any which way we approach this if `add-on === product` (not sure if the opposite is ever true). We are technically associating feature slugs with a product here.

Reply:
I see, it could be the Calypso equivalent of `wpcom_get_product_features()` (2cc77-pb/#plain) if we had access to the mapping in `WPCOM_Features`.

> I can work something else out if that method feels obtrusive or confusing in any way

Yeah I think that would be the right call. `useAddOnPurchaseStatus()` accepting two arguments would probably be what I'd go for. I just worry about `getFeatureSlug` leading folks who've never interacted with the Features setup down the wrong path. The fewer assumptions we can make the better.